compiler/src/main/java/com/juul/pommel/compiler/PommelProcessor.kt

```diff
@@ -67,108 +40,11 @@ class PommelProcessor : AbstractProcessor() {
 
     override fun process(annotations: Set<TypeElement>, roundEnv: RoundEnvironment): Boolean {
         roundEnv.findElementsAnnotatedWith<SoloModule>()
-            .map { it as TypeElement }
-            .mapNotNull { it.toPommelWriter() }
-            .forEach(this::writeSoloModule)
+            .mapNotNull { element ->
+                val codeGenerator = element.getCodeGenerator()
+                val params = codeGenerator?.validate(element, messager)
+                codeGenerator?.generate(checkNotNull(params), element)
+            }.forEach { file -> file.writeTo(filer) }
```

Comment:
If `getCodeGenerator()` returns null here, it's silently ignored. Should probably have an explicit error or warning message instead.